packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java

```diff
@@ -220,113 +220,121 @@ public void onFlutterViewDetached() {
+          Map<String, String> headers = (Map<String, String>) request.get("headers");
+          if (headers == null) {
+            headers = Collections.emptyMap();
+          }
+          loadUrl(webView, url, headers);
+          result.success(null);
+          break;
+        }
       case "updateSettings":
-        updateSettings(methodCall, result);
+        updateSettings(webView, methodCall.<Map<String, Object>>arguments());
+        result.success(null);
         break;
       case "canGoBack":
-        canGoBack(result);
+        result.success(canGoBack(webView));
         break;
       case "canGoForward":
-        canGoForward(result);
+        result.success(canGoForward(webView));
         break;
       case "goBack":
-        goBack(result);
+        goBack(webView);
+        result.success(null);
         break;
       case "goForward":
-        goForward(result);
+        goForward(webView);
+        result.success(null);
         break;
       case "reload":
-        reload(result);
+        reload(webView);
+        result.success(null);
         break;
       case "currentUrl":
-        currentUrl(result);
+        result.success(currentUrl(webView));
         break;
       case "evaluateJavascript":
-        evaluateJavaScript(methodCall, result);
+        evaluateJavaScript(webView, methodCall.<String>arguments(), result);
         break;
       case "addJavascriptChannels":
-        addJavaScriptChannels(methodCall, result);
+        addJavaScriptChannels(webView, methodCall.<List<String>>arguments());
+        result.success(null);
         break;
       case "removeJavascriptChannels":
-        removeJavaScriptChannels(methodCall, result);
+        removeJavaScriptChannels(webView, methodCall.<List<String>>arguments());
+        result.success(null);
         break;
       case "clearCache":
-        clearCache(result);
+        clearCache(webView);
+        result.success(null);
         break;
       case "getTitle":
-        getTitle(result);
+        result.success(getTitle(webView));
         break;
       case "scrollTo":
-        scrollTo(methodCall, result);
-        break;
+        {
+          Map<String, Object> request = methodCall.arguments();
+          scrollTo(webView, (int) request.get("x"), (int) request.get("y"));
+          result.success(null);
+          break;
+        }
       case "scrollBy":
-        scrollBy(methodCall, result);
-        break;
+        {
+          Map<String, Object> request = methodCall.arguments();
+          scrollBy(webView, (int) request.get("x"), (int) request.get("y"));
+          result.success(null);
+          break;
+        }
       case "getScrollX":
-        getScrollX(result);
+        result.success(getScrollX(webView));
         break;
       case "getScrollY":
-        getScrollY(result);
+        result.success(getScrollY(webView));
         break;
       default:
         result.notImplemented();
     }
   }
 
   @SuppressWarnings("unchecked")
```

Comment:
I believe this suppressions can now be removed as the casting of method call arguments has moved from this method to the `onMethodCall` method.